app/models/company.rb

```diff
@@ -2,4 +2,31 @@ class Company < ActiveRecord::Base
   validates :name, :city, presence: true
   validates :name, uniqueness: true
   has_many :jobs
+  has_many :contacts
+
+  def self.sort_by_city
+    order(:city)
+  end
+
+  def self.top_three_by_level_of_interest
+    format_results(extract_top_three)
```

Comment:
the actual formatting of your data should live in your View Helper. this method should simply return the actual objects. you could have another method that actually returns the average level of interest for a specific Company